.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -240,4 +242,63 @@ jobs:
       
       - name: Inspect image
         run: |
-          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
\ No newline at end of file
+          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
+
+  vulnerability_scan:
+    runs-on: ubuntu-latest
+    # if: inputs.publish
+    needs:
+    - docker_build
+    - prepare-metadata
+    steps:
+    - name: Create Trivy config
+      run: |
+        cat <<EOF > /tmp/trivy.yaml
+        format: sarif
+        exit-code: '1'
+        output: trivy-results.sarif
+        vulnerability:
+          ignore-unfixed: true  
+          type: 
+            - os
+            - library
+        severity: 
+          - CRITICAL
+          - HIGH
+        db:
+          repository: index.docker.io/aquasec/trivy-db:2
+        EOF
+    - name: Run Trivy vulnerability scanner
+      uses: aquasecurity/trivy-action@0.28.0
+      with:
+        image-ref: "${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}@${{ needs.docker_build.outputs.dockerhub-sha }}"
+        trivy-config: /tmp/trivy.yaml
+      env:
+        TRIVY_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+        TRIVY_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Upload Trivy scan results to GitHub Security tab
+      uses: github/codeql-action/upload-sarif@v3
+      if: always()
+      with:
+        sarif_file: 'trivy-results.sarif'
+
+
+  dockerfile_lint:
+    runs-on: ubuntu-latest
+      # if: inputs.publish
+    needs:
+      - docker_build
+      - prepare-metadata
+    steps:
+    - uses: actions/checkout@v3
+    - name: Dockerfile Linting
+      uses: hadolint/hadolint-action@v3.1.0
+      with:
+        output-file: hadolint-result.sarif
+        format: sarif
+        failure-threshold: error
+    - name: Upload linting results to GitHub Security tab
```

Comment:
I'm not a fan of uploading to result to GitHub Security tab, at least for linting, as it doesn't stop workflow from publishing if there are security issues.

Reply:
At first, we can ignore the error so that it won't block the team from having the image built (e.g https://github.com/babylonlabs-io/finality-provider/blob/main/.github/workflows/ci.yml#L17).
But we should encourage the team to fix the issues and then enforce the rules

Reply:
> I'm not a fan of uploading to result to GitHub Security tab, at least for linting, as it doesn't stop workflow from publishing if there are security issues.

hmm.. If you want it to stop publishing when there are issues, we can put this linting step in front of the build/publish step and requires it to pass in order to continue.

The uploading to Github Security Tab is just for clearer and easier view, i cannot see any connection between Uploading to Security tab and falling the workflow..

Reply:
> At first, we can ignore the error so that it won't block the team from having the image built (e.g https://github.com/babylonlabs-io/finality-provider/blob/main/.github/workflows/ci.yml#L17). But we should encourage the team to fix the issues and then enforce the rules

So.. basically the scanning result still being uploaded to the Github Security tab, but don't fail the pipeline when error found, right?

Reply:
> The uploading to Github Security Tab is just for clearer and easier view, i cannot see any connection between Uploading to Security tab and falling the workflow..

I don't think uploading to security tab and failing the workflow works at the same time as `Upload linting results to GitHub Security tab` step only run when the `Dockerfile Linting` step success. However, if we want the  `Dockerfile Linting` step to detect issues and stop the workflow, it won't reach the next step


```      
    - name: Dockerfile Linting
      uses: hadolint/hadolint-action@v3.1.0
      with:
        output-file: hadolint-result.sarif
        format: sarif
        failure-threshold: error
    - name: Upload linting results to GitHub Security tab
  ```

Reply:
Discussed offline that we don't want need the workflow to fail here